app/services/comments_notification_service.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
```

Comment:
Let's do:

Create a private method that find post that receive from params
`def find_post 
    @post = Post.find((params[:post_id]) 
end`

Then, it declare the before action only to initialize method
`before_action :find_post, only: %i[:initialize]`

Finally, In initialize method, you only make:
`def  initialize(params)
   @users = @post.users_in_comments 
   @owner = @post.user
end`